### ISSUE TITLE
 pagespeed_no_defer and pagespeed_lsc_url break HTML5 validation

### ISSUE THREAD

**Issue by GoogleCodeExporter** [pagespeed/mod_pagespeed#584]
```
mod_pagespeed alters script and css tags to have the above tags, which break 
HTML5 validation (see 
http://validator.w3.org/check?uri=https://workbenchwarriors.com/&charset=%28dete
ct+automatically%29&doctype=HTML5&group=0 for an example).

These attributes would be better added as data-pagespeed-no-defer and 
data-pagespeed-lsc-url respectively, to pass HTML5 validation (see described 
here: http://html5doctor.com/html5-custom-data-attributes/) 



```

Original issue reported on code.google.com by `bar...@shrikeh.net` on 7 Dec 2012 at 11:57

**Comment by slayer1ss** [comment]
Hi, i am having same issue with prioritize_critical_css when i activate it pagespeed adds < script pagespeed_no_defer="">critical_css_loader script stuff</ script> to the bottom of my page and it breaks the validation, will there be a fix for this soon?

**Comment by jmaessen** [comment]
No.  Moving non-critical CSS to the bottom of the page is how prioritize_critical_css functions – there's no standards-compliant way we know of to lazy load the CSS without adding additional latency to your page.  Given the choice, this filter has been forced to opt for a working solution rather than a strict standards-compliant one.

That said, we've change the non-standard pagespeed attributes to be spec-compliant; in the next release you should be able to use data-pagespeed-no-defer and pagespeed will indeed insert data-pagespeed-lsc-url.  (This is currently available in head if you build from source.)

My suggestion: disable prioritize_critical_css and enable pedantic when you are running a validator.  We suggest that you do the reverse in day-to-day operation.

**Comment by slayer1ss** [comment]
Just to be clear, i am perfectly happy with how prioritize_critical_css functions and wouldnt live without it... The problem is pagespeed-no-defer tag which is being add automaticly, so if i understand you correctly on next release that pagespeed-no-defer tag which was added by mps will automaticly change to data-pagespeed-no-defer and there wont be a problem anymore am i right?

Is this change included in latest beta release?

**Comment by crowell** [comment]
@slayer1ss the change here https://github.com/pagespeed/mod_pagespeed/commit/ab760635d77ee325c02da485c5abbcf09a6affa6 as of now isn't included in the latest beta release.

**Comment by slayer1ss** [comment]
@crowell thank you very much for that information, it actually both answers my question about if i got it right and tells me that it is not ready yet :) I dont want to take any more of your time but do you guys have a timetable for this update?
